app/Http/Controllers/ThreadController.php

```diff
@@ -2,22 +2,28 @@
 
 namespace App\Http\Controllers;
 
+use App\Filters\ThreadFilters;
 use App\Http\Requests\ThreadRequest;
 use App\Models\Channel;
 use App\Models\Thread;
 
 class ThreadController extends Controller
 {
+
     /**
-     * Display a listing of the resource.
-     *
-     * @param \App\Models\Channel $channel
+     * @param Channel|null $channel
+     * @param ThreadFilters $filters
      *
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\JsonResponse
      */
-    public function index(?Channel $channel = null)
+    public function index(?Channel $channel = null, ThreadFilters $filters)
     {
-        return ! is_null($channel) ? $channel->threads->all() : Thread::all();
+        if (! is_null($channel)) {
+            $threads = $channel->threads->all();
+        } else {
+            $threads = Thread::filter($filters)->get();
+        }
+        return response()->json($threads);
     }
 
```

Comment:
when we have an if else statement we usually use a ternary operator to slim things down.